### ISSUE TITLE
Login loop when using Plex authentication.

### ISSUE THREAD

**Issue by knightowl1000** [tidusjar/Ombi#2755]
**Describe the bug**
Login loop when using Plex authentication.

**To Reproduce**
1. Go to Ombi homepage
2. Click on Sign in With Plex
3. Authenticate to Plex

**Expected behavior**
Once Signed into the plex auth, Ombi should log into the portal.

**Screenshots**
![ezgif-2-7dd627373c3f](https://user-images.githubusercontent.com/39104828/51557615-0a8a7900-1e43-11e9-856e-8c39e39cbd7a.gif)

**Logs (Logs directory where Ombi is located)**
Log files only show issues connecting to themoviedb

**Desktop (please complete the following information):**
- Docker Container

**Ombi Version (please complete the following information):**
 - Version 3.0.4119
- Media Server - Plex

**Additional context**
Users inside and outside of the network are facing the same issue. Sometimes clearing browser cache works, other times it doesn't. It happens randomly, but more often than not recently.

**Comment by tidusjar** [comment]
Can you please install the latest develop and try again following the below
This will be for build 3.0.4156

Are there any errors in your browser console? F12.

If so can you press the network tab and then refresh the page and you should see something like
![image](https://cloud.githubusercontent.com/assets/6642220/23088047/eac44abc-f570-11e6-9f23-d2eb022ed059.png)


If you click the red text and then the preview tab like the image above and post the result?

**Comment by knightowl1000** [comment]
Hi @tidusjar - I'm on 3.0.4157 develop branch right now. I'm not running into the login loop. I've requested that people using it try and see if they're running into the problem and will get back to you.

**Comment by tidusjar** [comment]
Yeah ideally I need the error from the browsers console 😢

**Comment by knightowl1000** [comment]
So far no luck >.< I'll keep on trying.

**Comment by SlackMaster** [comment]
I'm having this issue. I'm kind of a noob so you'll have to walk me through what's required to help solve this issue. 

Latest stable release installed on Ubuntu 18.04
Have tried different browsers, clearing cache, removing the authorized devices in Plex, Inside of network, outside of network, using the local IP and port, and using the reverse proxy address. None of it appears to work.

As a side note, I can't login with my local admin either, keeps telling me it's the wrong password. Not sure if that's related or not, but I'm almost certain the local password was working yesterday. Also, earlier today I tried using the Ombi app for iOS and couldn't get it to work. Again, not sure if that's related but there's that.

Here's what I get from the browser.

![screen shot 2019-01-31 at 2 04 08 pm](https://user-images.githubusercontent.com/353988/52078611-9ac26f80-2561-11e9-85c1-2df5decc350a.png)

**Comment by tidusjar** [comment]
From your error it seems that the user you attempted to login with doesn't exist in ombi. I should improve the feedback from that.

**Comment by SlackMaster** [comment]
I tried three different Plex accounts. The Plex website always indicates the login is success ful and when I check the DB, the Plex accounts were imported correctly. 

I did a full reinstall and didn’t enable OAuth, just imported users and it’s been working so far. 

I thought you had to check the box to allow OAuth to allow login from Plex?

**Comment by tidusjar** [comment]
The oauth button makes the login with Plex available on the login page

**Comment by jeansebb** [comment]
I've got a loop login issue with the following console error : error:
error: "SQLite Error 5: 'database is locked'."

**Comment by simonk83** [comment]
Also having this issue...

**Comment by prisoner0627** [comment]
Also having login issues running Ombi on Docker on unRAID server. I have tried switching to Dev and deleting Schedules.db but still not able to login with any account.

**Comment by SlackMaster** [comment]
At least for my case, I found that the date/time settings for the server were not correct and that seemed to be affecting the login. Soon as I updated those setting, which updated the docker container, the login started working like it was supposed to. Hope this helps for others.

**Comment by douginoz** [comment]
Same problem here.  Kubuntu 18.10 and Kubuntu 19.04.  Latest stable release of mbi  ( I think, not sure how to check).  Firefox console logs are:
`Could not read chrome manifest 'file:///usr/lib/firefox/chrome.manifest'.
1560371945097	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: mozillaAddons
1560371945097	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: resource://pdf.js/
1560371945097	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid extension permission: about:reader*
1560371945901	addons.webextension.plasma-browser-integration@kde.org	WARN	Loading extension 'plasma-browser-integration@kde.org': Reading manifest: Error processing background.persistent: Event pages are not currently supported. This will run as a persistent background page.
1560371945903	addons.webextension.plasma-browser-integration@kde.org	WARN	Loading extension 'plasma-browser-integration@kde.org': Reading manifest: Error processing key: An unexpected property was found in the WebExtension manifest.
1560371945909	addons.webextension.chrome-gnome-shell@gnome.org	WARN	Loading extension 'chrome-gnome-shell@gnome.org': Reading manifest: Error processing background.persistent: Event pages are not currently supported. This will run as a persistent background page.
1560371945913	addons.webextension.chrome-gnome-shell@gnome.org	WARN	Loading extension 'chrome-gnome-shell@gnome.org': Reading manifest: Error processing externally_connectable: An unexpected property was found in the WebExtension manifest.
Unkown category for SetEventRecordingEnabled.
The character encoding of the HTML document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the page must be declared in the document or in the transfer protocol. extension.html
The character encoding of the HTML document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the page must be declared in the document or in the transfer protocol. extension.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. update.js:26
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist.
start.js:92
Log warning: The log 'Services.Common.RESTRequest' is configured to use the preference 'services.common.log.logger.rest.request' - you must adjust the level by setting this preference, not by using the level setter Log.jsm:20
TypeError: l.usernameField is null
LoginHelper.jsm:137:1
NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS: [JavaScript Error: "l.usernameField is null" {file: "resource://gre/modules/LoginHelper.jsm" line: 137}]'[JavaScript Error: "l.usernameField is null" {file: "resource://gre/modules/LoginHelper.jsm" line: 137}]' when calling method: [nsILoginManagerStorage::addLogin] LoginManager.jsm:296
Use of the orientation sensor is deprecated. app.js:1615:1356
Use of the motion sensor is deprecated. app.js:1615:1356
Invalid chrome URI: /
Password fields present on an insecure (http://) page. This is a security risk that allows user login credentials to be stolen.
login
Use of the orientation sensor is deprecated. app.js:1615:1356
Use of the motion sensor is deprecated. app.js:1615:1356
this._box is null browser-sidebar.js:328
Use of the orientation sensor is deprecated. app.js:1615:1356
Use of the motion sensor is deprecated. app.js:1615:1356
Loading failed for the <script> with source “https://www.google-analytics.com/analytics.js”. login:1:1
Content Security Policy: Ignoring ‘x-frame-options’ because of ‘frame-ancestors’ directive.
Loading failed for the <script> with source “https://connect.facebook.net/en_US/fbevents.js”. auth:1:1
Loading failed for the <script> with source “https://platform.twitter.com/oct.js”. auth:1:1
Loading failed for the <script> with source “https://googleads.g.doubleclick.net/pagead/viewthroughconversion/877994546/?random=1560372013458&cv=9&fst=1560372013458&num=1&guid=ON&resp=GooglemKTybQhCsO&u_h=1080&u_w=1920&u_ah=1080&u_aw=1920&u_cd=24&u_his=1&u_tz=-420&u_java=false&u_nplug=1&u_nmime=2&gtm=2wg5t2&sendb=1&frm=0&url=https%3A%2F%2Fapp.plex.tv%2Fauth&ref=http%3A%2F%2F192.168.1.39%3A5000%2Flogin&tiba=Plex&async=1&rfmt=3&fmt=4”. auth:1:1
Loading failed for the <script> with source “https://googleads.g.doubleclick.net/pagead/viewthroughconversion/973693106/?random=1560372013471&cv=9&fst=1560372013471&num=1&guid=ON&resp=GooglemKTybQhCsO&u_h=1080&u_w=1920&u_ah=1080&u_aw=1920&u_cd=24&u_his=1&u_tz=-420&u_java=false&u_nplug=1&u_nmime=2&gtm=2wg5t2&sendb=1&frm=0&url=https%3A%2F%2Fapp.plex.tv%2Fauth&ref=http%3A%2F%2F192.168.1.39%3A5000%2Flogin&tiba=Plex&async=1&rfmt=3&fmt=4”. auth:1:1
[Exception... "Favicon at "https://assets.plex.tv/deploys/auth/env-294593abcd4dd15ff4085183fdee3f83/3.99.0-a8effbc/favicon.ico" failed to load: Forbidden."  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource:///modules/FaviconLoader.jsm :: onStopRequest :: line 199"  data: no]
[Exception... "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIDOMWindowUtils.removeSheetUsingURIString]"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource://gre/modules/ExtensionCommon.jsm :: runSafeSyncWithoutClone :: line 65"  data: no]
TypeError: l.usernameField is null
LoginHelper.jsm:137:1
NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS: [JavaScript Error: "l.usernameField is null" {file: "resource://gre/modules/LoginHelper.jsm" line: 137}]'[JavaScript Error: "l.usernameField is null" {file: "resource://gre/modules/LoginHelper.jsm" line: 137}]' when calling method: [nsILoginManagerStorage::addLogin] LoginManager.jsm:296
Key event not available on some keyboard layouts: key=“r” modifiers=“accel,alt” id=“key_toggleReaderMode” browser.xul
Key event not available on some keyboard layouts: key=“i” modifiers=“accel,alt,shift” id=“key_browserToolbox” browser.xul
Promise resolved after context unloaded
content-script.js:64
Promise resolved after context unloaded
preload.js:374
Content Security Policy: Ignoring ‘x-frame-options’ because of ‘frame-ancestors’ directive.
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”).
2755:1:1
This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://developer.mozilla.org/docs/Mozilla/Performance/ScrollLinkedEffects for further details and to join the discussion on related tools and features! 2755
update.locale file doesn't exist in either the application or GRE directories UpdateUtils.jsm:138
<unavailable> app.js:1530:36995
<unavailable>
app.js:1530:36995
<unavailable> app.js:1530:36995
<unavailable> app.js:1530:36995
<unavailable> app.js:1530:36995
<unavailable> app.js:1530:36995
<unavailable>
app.js:1530:36995
<unavailable> app.js:1530:36995
<unavailable>
app.js:1530:36995
<unavailable> app.js:1530:36995`
`

**Comment by douginoz** [comment]
Seems to be fixed in latest develop branch.  See https://github.com/tidusjar/Ombi/issues/2994

**Comment by 2fst4u** [comment]
I still continue to get this issue. The only way to fix it is to restart Ombi.

This happens daily, basically like clockwork. None of my user can use the application until I manually reboot it.

The time on my server is correct, I just check it. I am on the develop branch and I update each time in the hopes that it is fixed but it doesn't seem to be.

**Comment by Alenux55** [comment]
> I still continue to get this issue. The only way to fix it is to restart Ombi.
> 
> This happens daily, basically like clockwork. None of my users can use the application until I manually reboot it.
> 
> The time on my server is correct, I just checked it. I am on the develop branch and I update each time in the hopes that it is fixed but it doesn't seem to be.

I'm seeing this as well, however a restart does not fix it. Here's my console:
![Reload_issue](https://user-images.githubusercontent.com/24886818/69189464-45952180-0ad3-11ea-8b9a-ff8db4591355.png)

I'm running this on Ubuntu with the apt-repo, using the latest Obmi version (3.0.4892). Logs contain no mention of the error it seems

**Comment by Alenux55** [comment]
I ended up restarting from scratch to fix this issue. Was probably a database import issue.

**Comment by gnade** [comment]
This is also occurring for me. Was happening on an Ubuntu docker install, moved it to a windows host. Same behaviour.
Local account logs in fine, plex auth loops. 

Console view in chrome shows Failed to load resource: the server responded with a status of 401 (Unauthorized) over and over again.

Version
3.0.4892 Master

**Comment by th0ul0w88** [comment]
I'm having this same issue... I've tried multiple versions of Ombi.

Currently on 
Omb -- 4.0.177.0
Plex -- 4.12.3
Windows 10 Pro

I've tried Chrome as well as Edge browser

**Comment by jeansebb** [comment]
Try the MySQL option, seems like SQLite us not well suited for all everything Ombi can do.

**Comment by gnade** [comment]
Since this thread is closed it is not likely to get any attention.

**Comment by tidusjar** [comment]
Yeah sorry I don't. 

I suggest you switch to mysql, theres a guide on the wiki relating to this. It seems that we have hit the sqlite limit.

**Comment by KrunchMuffin** [comment]
So, I just switched to mariadb and this is still happening. I will get console errors when back at my desktop.

**Comment by KrunchMuffin** [comment]
3.0.4892

With PLex Login
![image](https://user-images.githubusercontent.com/1037197/77497512-de420200-6e23-11ea-8bf8-3e421b004c94.png)

With Ombi login
![image](https://user-images.githubusercontent.com/1037197/77497547-f580ef80-6e23-11ea-8a21-fd101e9e5bbb.png)

**Comment by gnade** [comment]
I switched mine to MySQL and have not encountered the issue again. I wouldn't say my library is large by any stretch, disappointing the out of the box experience seems to be limited to a very small install size. 

The DB technology shouldn't matter too much, but maybe try MySQL. I never tried mariana

**Comment by KrunchMuffin** [comment]
It's a fork of MySQL. I think if he used WAL https://www.sqlite.org/wal.html for SQLite that it would not cause lock contention. But maybe he has tried that already. Sonarr and Radarr use it.
At any rate, my issue does not appear to be with locking.

**Comment by jtilles** [comment]
I recently ran into this problem.  I am using a reveres proxy via [NGINX Proxy Manager](https://nginxproxymanager.com/) and ran into this loop when I had enabled *Websockets Support*.  When I turned that off, problem was fixed.  I am using the LinuxServer.io Docker container to run Ombi.

![OMBIreverseProxy](https://user-images.githubusercontent.com/10439896/77759104-84e7f780-700a-11ea-8ef3-1531f265ec57.png)
